### DISCUSSION TITLE
How to create style consistency with ChakraUi

### DISCUSSION THREAD

**Discussion by florianmartens** [chakra-ui/chakra-ui#4656] [Q&A] [answered]
When working with Chakra (I'm in ❤️  with the library), I often find myself wondering how to create consistent styles. A good example is that there is no way to introduce muted text. Let's say I want to do this:

`<Text color="gray.400">Muted</Text>`

In a big app, I'll quickly end up with hundreds of places where I've defined a somewhat magic string of "gray.400" and I have to remember that value. I've experimented with adding a second theming layer that injects defaults for e. g. `textMuted` into my app. But I've wondered if there is a better way. Would love to hear common approaches to this.

**Comment by aaronkik** [comment, marked as answer]
I've actually had to do this, I use the [Chakra factory function](https://chakra-ui.com/docs/features/chakra-factory) and created a file called `MutedText.tsx` with the contents being: 
```js
import { chakra, TextProps } from '@chakra-ui/react';

const MutedText = chakra<'p', TextProps>('p', {
  baseStyle: {
    opacity: 0.8,
    // Any other styles ...
  },
});

export default MutedText;
```

When it comes to using it:
```js
<MutedText fontSize='sm'>Muted</MutedText>
```
